sea-orm-codegen/src/entity/name_resolver.rs

```diff
@@ -0,0 +1,45 @@
+use heck::{CamelCase, SnakeCase};
+use inflection::singular;
+
+#[derive(Clone, Debug)]
+pub struct NameResolver {
+    singularize: bool
+}
+
+impl NameResolver {
+    pub fn new(singularize: bool) -> Self {
+        Self {
+            singularize
+        }
+    }
+    
+    pub fn resolve_module_name(&self, name: &str) -> String {
+        let name = name.to_snake_case();
+
+        if self.singularize {
+            singular(name)
+        } else {
+            name
+        }
+    }
+
+    pub fn resolve_entity_name(&self, name: &str) -> String {
+        let name = name.to_camel_case();
+        
+        if self.singularize {
+            singular(name)
+        } else {
+            name
+        }
+    }
+
+    pub fn resolve_relation_name(&self, name: &str) -> String {
+        let name = name.to_camel_case();
+
+        if self.singularize {
+            singular(name)
+        } else {
+            name
+        }
+    }
```

Comment:
Hey @negezor, thanks for the hard rebase (it's hard as this PR is far behind from master).

I wonder why we need three different methods? I think we can merge it and have two methods? One for snake_case and the other for camel_case.

Reply:
A fair remark, but I proceeded from the clear goals of the method. I can make changes if necessary.